src/NuGet.Core/NuGet.Build.Tasks.Pack.Library/PackTask.cs

```diff
@@ -79,9 +84,12 @@ public override bool Execute()
         {
             var request = GetRequest();
             var logic = PackTaskLogic;
-
+            PackageBuilder packageBuilder = null;
             var packArgs = logic.GetPackArgs(request);
-            var packageBuilder = logic.GetPackageBuilder(request);
+            if (string.IsNullOrEmpty(request.NuspecFile))
```

Comment:
yes, the logic here is that :

if a nuspec file has been specified, then we don't need to build a package builder here as packing from nuspec doesn't rely on msbuild inputs.

if a nuspec file hasn't been specified, then we create packagebuilder object here and use it downstream.

both are valid codepaths.